perf/fiotest.py

```diff
@@ -44,16 +46,34 @@ def setUp(self):
         """
         Build 'fio'.
         """
-        tarball = self.fetch_asset('http://brick.kernel.dk/snaps/fio-2.1.10.tar.gz')
+        sm = SoftwareManager()
+        dist = distro.detect()
+        if 'Ubuntu' in dist.name:
+            deps = ['libaio1', 'libaio-dev', 'bzip2']
+        else:
+            deps = ['libaio', 'libaio-devel', 'bzip2']
+        for package in deps:
+            if not sm.check_installed(package) and not sm.install(package):
+                self.error(
+                    '%s is needed for the test to be run !!' % (package))
+        tarball = self.fetch_asset(
+            'http://brick.kernel.dk/snaps/fio-git-latest.tar.gz')
         archive.extract(tarball, self.srcdir)
-        fio_version = os.path.basename(tarball.split('.tar.')[0])
-        self.srcdir = os.path.join(self.srcdir, fio_version)
+        self.srcdir = os.path.join(self.srcdir, 'fio')
+        os.chdir(self.srcdir)
+        process.run('./configure', ignore_status=True, sudo=True)
         build.make(self.srcdir)
 
     def test(self):
         """
         Execute 'fio' with appropriate parameters.
         """
+        self.testdir = '/data1'
+        self.scrtchdir = '/scratch'
```

Comment:
perhaps, the test directories are created in root '/' and it may use the disk space of root device. as long as root disk is more than 5G than no space issue